src/connect/ConnectWizard.ts

```diff
@@ -147,32 +89,132 @@ export class ConnectWizard {
+        }
+        catch (error) {
+            // We want to recover if for some reason the user isn't able to list namespaces.
+            this._logger.warning(`Failed to list namespaces`, error);
+        }
+        if (namespaces != null && !namespaces.includes(targetResourceNamespace)) {
+            // In practice, this error should happen rarely, as the K8s cluster explorer only allows interaction with resources from the current cluster
+            throw new Error(`Failed to find the namespace '${targetResourceNamespace}' in cluster '${currentContext.cluster}'`);
+        }
+    }
+
+    private async handlePodResourceType(targetResourceName: string, targetResourceNamespace: string, currentContext: IKubeconfigEnrichedContext, bridgeClient: any, kubectlClient: any) {
+        // We skip the isolation step when debugging pods
+        this.NumberOfSteps = 3;
+        await MultiStepInput.runAsync(async (input) => {
+            await this.prevalidateAsync(ResourceType.Pod, targetResourceName, targetResourceNamespace, currentContext, bridgeClient, kubectlClient, input);
+            let resourceNameToPersist: string = targetResourceName;
+            // Because the specific pod name changes, we persist only the first segments of the name, e.g. <deployment name>-<deployment guid>-.
+            const split: string[] = targetResourceName.split(`-`);
+            if (split.length > 1) {
+                // Delete last segment
+                split.splice(-1, 1);
+                resourceNameToPersist = `${split.join(`-`)}`;
+            }
+
+            this._result.resourceName = resourceNameToPersist;
+            this._result.targetCluster = currentContext.cluster;
+            this._result.targetNamespace = targetResourceNamespace;
+            this._result.resourceType = ResourceType.Pod;
+
+            return await this.getContainerSelection(targetResourceName, ResourceType.Pod, kubectlClient);
+        });
+    }
+
+    private async handleServiceResourceType(targetResourceName: string, targetResourceNamespace: string, currentContext: IKubeconfigEnrichedContext, bridgeClient: any, kubectlClient: any) {
+        await MultiStepInput.runAsync(async (input) => {
+            await this.prevalidateAsync(ResourceType.Service, targetResourceName, targetResourceNamespace, currentContext, bridgeClient, kubectlClient, input);
+            return this.pickServiceAsync(input, currentContext, ResourceType.Service);
         });
-        return this._isWizardComplete ? this._result as IWizardOutput : null;
     }
 
+
+    private async getContainerSelection(targetResourceName: string, targetResourceType: ResourceType, kubectlClient: any): Promise<(input: MultiStepInput) => Promise<any>> {
+        // get the list of containers
+        const containersList = await kubectlClient.getContainerNames(targetResourceName, this._result.targetNamespace);
+
+        if (containersList?.length > 1) {
```

Comment:
It looks like we missed this when we looked at it together earlier, but I think a separate `if` block to handle the error case explicitly would be clearer:
```typescript
if (containersList === null) {
    // There was an error retrieving containers for this pod.
    this._result.containerName = undefined;
    return (input: MultiStepInput) => this.inputPortsAsync(input, targetResourceType);
}
```
And then we can delete all the `?` characters for the remainder of the function.